### PR TITLE
GUI cold signing

### PR DESCRIPTION
**Pull request by Jaqueeee**
also fixes https://github.com/monero-project/monero-core/issues/389

### CODE REVIEW

**Comment by fluffypony** [comment]
Please rebase

**Comment by Jaqueeee** [comment]
Don't merge this yet. have some conflicts to resolve.

**Comment by Jaqueeee** [comment]
Rebased.

File: src/wallet/api/unsigned_transaction.cpp [hunk 1]
```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2014-2016, The Monero Project
```
Comment (moneromooo-monero) [inline src/wallet/api/unsigned_transaction.cpp]: I guess it's 2017 now ^_^

File: src/wallet/api/pending_transaction.cpp [hunk 2]
```diff
@@ -77,19 +77,39 @@ std::vector<std::string> PendingTransactionImpl::txid() const
...
 {
 
     LOG_PRINT_L3("m_pending_tx size: " << m_pending_tx.size());
 
     try {
+      // Save tx to file
+      if (!filename.empty()) {
+        boost::system::error_code ignore;
+        bool tx_file_exists = boost::filesystem::exists(filename, ignore);
+        if(tx_file_exists && !overwrite){
```
Comment (Jaqueeee) [inline src/wallet/api/pending_transaction.cpp]: Added optional overwrite here after moo's review.

**Pull request merged**